iceoryx_posh/test/moduletests/test_roudi_portpool.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/roudi/port_pool_data.hpp"
+#include "iceoryx_posh/internal/runtime/node_data.hpp"
+#include "iceoryx_posh/roudi/port_pool.hpp"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using ::testing::Return;
+
+namespace example
+{
+namespace test
+{
+class PortPool_test : public Test
+{
+  public:
+    iox::roudi::PortPoolData* portPoolData = new iox::roudi::PortPoolData{};
```

Comment:
You need to delete this, otherwise there is a memleak

Reply:
Update:
I was wondering why PortPoolData was created on heap, i think this is not necessary here. We should avoid heap in test too if possible.
My advise would be to restructure the test:

1. remove the placement new for PortPoolData and add it directly to the test class as member
2. use the PortPoolData member to add a new PortPool member with the name `sut` to make clear that this is your "system under test". Use the previously created PortPoolData member to initialize it.
3. use `sut` in your test case functions to fill the vector and test it.

Reply:
done